star.wdl

```diff
@@ -20,6 +20,70 @@ version 1.0
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+task GenomeGenerate {
+    input {
+        String genomeDir = "STAR_index"
+        File referenceFasta
+        File? referenceGtf
+        Int? sjdbOverhang
+
+        Int threads = 4
+        String memory = "60G"
+        String dockerImage = "quay.io/biocontainers/star:2.7.3a--0"
+    }
+
+    command {
+        set -e
+        mkdir -p "$(dirname ~{genomeDir})"
+        STAR \
+        --runMode genomeGenerate \
+        --runThreadN ~{threads} \
+        --genomeDir ~{genomeDir} \
+        --genomeFastaFiles ~{referenceFasta} \
+        ~{"--sjdbGTFfile " + referenceGtf} \
+        ~{"--sjdbOverhang " + sjdbOverhang}
+    }
+
+    output {
+        File chrLength = "~{genomeDir}/chrLength.txt"
+        File chrNameLength = "~{genomeDir}/chrNameLength.txt"
+        File chrName = "~{genomeDir}/chrName.txt"
+        File chrStart = "~{genomeDir}/chrStart.txt"
+        File genome = "~{genomeDir}/genome.txt"
+        File genomeParameters = "~{genomeDir}/genomeParameters.txt"
+        File sa = "~{genomeDir}/SA"
+        File saIndex = "~{genomeDir}/SAindex"
+        File? exonGeTrInfo = "~{genomeDir}/exonGeTrInfo.tab"
+        File? exonInfo = "~{genomeDir}/exonInfo.tab"
+        File? geneInfo = "~{genomeDir}/geneInfo.tab"
+        File? sjdbInfo = "~{genomeDir}/sjdbInfo.txt"
+        File? sjdbListFromGtfOut = "~{genomeDir}/sjdbList.fromGTF.out.tab"
+        File? sjdbListOut = "~{genomeDir}/sjdbList.out.tab"
+        File? transcriptInfo = "~{genomeDir}/transcriptInfo.tab"
+        Array[File] starIndex = select_all([chrLength, chrNameLength, chrName, chrStart, genome, genomeParameters,
+                                            sa, saIndex, exonGeTrInfo, exonInfo, geneInfo, sjdbInfo, sjdbListFromGtfOut,
+                                            sjdbListOut, transcriptInfo])
+    }
+
+    runtime {
+        cpu: threads
```

Comment:
runtime_minutes?
On 4 threads a human genome should succeed within 2 hours.
The size function should be able to put out the size in GB
so `size(referenceFasta, "G") * 120 / runThreadN` ? i.e. 2 hours per GB on one thread.
Now I am thinking. We could even set a generic `Int minutesPerGB` at the top of each task.
So the formula below becomes `size(referenceFasta, "G") * minutesPerGb /runThreadN` .